number_theory/modular.h

```diff
@@ -111,6 +112,120 @@ class Modular {
+  }                                                       \
+  template <CONCEPT M, std::convertible_to<M> T>          \
+  M operator OP(const M &lhs, const T &rhs) {             \
+    return lhs.METHOD(static_cast<M>(rhs));               \
+  }                                                       \
+  template <CONCEPT M, std::convertible_to<M> T>          \
+  M operator OP(const T &lhs, const M &rhs) {             \
+    return static_cast<M>(lhs).METHOD(rhs);               \
+  }
+
+// Overloads an inplace operator with a class method
+#define OVERLOAD_INPLACE_OPERATOR(CONCEPT, OP, METHOD) \
+  template <CONCEPT M, std::convertible_to<M> T>       \
+  M &operator OP(M &lhs, const T &rhs) {               \
+    lhs = lhs.METHOD(static_cast<M>(rhs));             \
+    return lhs;                                        \
+  }
+
+// Overloads a pair of comparison operator with a class method
+#define OVERLOAD_COMPARISON_OPERATOR(CONCEPT, OP, OP_NEG, METHOD) \
+  template <CONCEPT M>                                            \
+  bool operator OP(const M &lhs, const M &rhs) {                  \
+    return lhs.METHOD(rhs);                                       \
+  }                                                               \
+  template <CONCEPT M, std::convertible_to<M> T>                  \
+  bool operator OP(const M &lhs, const T &rhs) {                  \
+    return lhs.METHOD(static_cast<M>(rhs));                       \
+  }                                                               \
+  template <CONCEPT M, std::convertible_to<M> T>                  \
+  bool operator OP(const T &lhs, const M &rhs) {                  \
+    return static_cast<M>(lhs).METHOD(rhs);                       \
+  }                                                               \
+  template <CONCEPT M>                                            \
+  bool operator OP_NEG(const M &lhs, const M &rhs) {              \
+    return !lhs.METHOD(rhs);                                      \
+  }                                                               \
+  template <CONCEPT M, std::convertible_to<M> T>                  \
+  bool operator OP_NEG(const M &lhs, const T &rhs) {              \
+    return !lhs.METHOD(static_cast<M>(rhs));                      \
+  }                                                               \
+  template <CONCEPT M, std::convertible_to<M> T>                  \
+  bool operator OP_NEG(const T &lhs, const M &rhs) {              \
+    return !static_cast<M>(lhs).METHOD(rhs);                      \
+  }
+
+// Overloads an unary operator with a class method
+#define OVERLOAD_UNARY_OPERRATOR(CONCEPT, OP, METHOD) \
+  template <CONCEPT M>                                \
+  M operator OP(const M &x) {                         \
+    return x.METHOD();                                \
+  }
+
+// Overload an increment/decrement operator with a class method
+#define OVERLOAD_INCDEC_OPERRATOR(CONCEPT, OP, METHOD) \
```

Comment:
I wonder if there exists a better name for those postfix/prefix operators instead of `INCDEC`.

Reply:
Actually I am not sure.
I was refering to this: https://en.cppreference.com/w/cpp/language/operator_incdec
Notice that the url is named `operator_incdec`.